autonomi/src/client/data_map_restoration.rs

```diff
@@ -0,0 +1,150 @@
+    ///   * repeat the above two steps if the recovered data_map contains chiled
+    ///   * fetch the leftover content chunks to compose the file
+    pub(crate) async fn restore_data_map_from_chunk(
+        &self,
+        data_map_chunk: &DataMapChunk,
+    ) -> Result<DataMap, GetError> {
+        let mut data_map_bytes = data_map_chunk.0.value().clone();
+
+        // In case the input is the new data_map root, restore it further,
+        // before going into the further recursive.
+        if let Ok(data_map) = rmp_serde::from_slice::<DataMap>(&data_map_bytes) {
+            info!("Restoring from new root data_map:\n{data_map:?}");
+            let file_data_map = self.fetch_new_data_map(&data_map)?;
+
+            info!("Fetched file data_map of new version: \n{file_data_map:?}");
+            return Ok(file_data_map);
+        }
+
+        loop {
+            // The data_map_bytes could be an Archive, we shall return earlier for that case
+            match Self::deserialize_data_map(&data_map_bytes) {
+                Ok(mut data_map) => {
+                    info!("Restoring from data_map:\n{data_map:?}");
+                    if !data_map.is_child() {
+                        return Ok(data_map);
+                    }
+                    data_map.child = None;
+                    data_map_bytes = self.fetch_from_data_map(&data_map).await?;
+                }
+                Err(e) => {
+                    info!("Failed to deserialize data_map_bytes: {e:?}");
+                    return Err(GetError::Configuration(format!("{e:?}")));
+                }
+            }
+        }
+    }
+
+    /// Fetch the file data_map from the root one using lazy evaluation.
+    /// Chunks are only fetched from the network when actually needed by get_root_data_map.
+    fn fetch_new_data_map(&self, data_map: &DataMap) -> Result<DataMap, GetError> {
+        let total_chunks = data_map.infos().len();
+        #[cfg(feature = "loud")]
+        println!(
+            "Setting up lazy chunk fetching for {total_chunks} chunks of data_map {data_map:?}"
+        );
+        debug!("Setting up lazy chunk fetching for {total_chunks} chunks of data_map {data_map:?}");
+
+        // Create a closure that fetches chunks on-demand
+        let client = self.clone();
+        let mut chunk_fetcher = move |xor_name: XorName| -> Result<Bytes, self_encryption::Error> {
+            let chunk_addr = ChunkAddress::new(xor_name);
+
+            // Use tokio::task::spawn_blocking to handle the async operation in a sync context
+            let fetch_result = tokio::task::block_in_place(|| {
+                tokio::runtime::Handle::current()
+                    .block_on(async { client.chunk_get(&chunk_addr).await })
+            });
+
+            match fetch_result {
+                Ok(chunk) => {
+                    #[cfg(feature = "loud")]
+                    println!("Successfully fetched chunk for XorName: {xor_name:?}");
```

Comment:
Is it possible to just use "address" rather than "XorName" when surfacing to the user?